### ISSUE TITLE
Unknown setting custom_url on 22.11.1

### ISSUE THREAD

**Issue by ramazanpolat** [metrico/urleng#1]
Hello there.

Great idea of utilizing url table function : )

There is one small issue tho, on the main page, it says we can do this: 

`SET custom_url = 'https://urleng.com/4c5u94cw592bsq9wypba'`

which is actually not possible.

```sql
SET custom_url = 'https://urleng.com/4c5u94cw592bsq9wypba'

Query id: 88fbda2b-f2ad-4b99-ba1c-39e2b5f141e3


0 rows in set. Elapsed: 0.001 sec. 

Received exception from server (version 22.11.1):
Code: 115. DB::Exception: Received from localhost:9000. DB::Exception: Unknown setting custom_url. (UNKNOWN_SETTING)
```

But this works:

```sql
:) SET param_url = 'https://urleng.com/abc'

SET param_url = https://urleng.com/abc

Query id: 1a4c6ab9-e3bf-436f-9723-335a9d55f7c8

Ok.

0 rows in set. Elapsed: 0.001 sec. 

:) INSERT INTO FUNCTION url({url:String}, JSONEachRow, 'key String, value UInt64') VALUES ('hello', 1)

INSERT INTO FUNCTION url({url:String}, JSONEachRow, 'key String, value UInt64') FORMAT Values

Query id: c6e52e51-963b-4233-aa65-e0792e29b3e1

Ok.

1 row in set. Elapsed: 0.802 sec. 

:) SELECT * FROM url({url:String}, JSONEachRow, 'key String, value UInt64')

SELECT *
FROM url({url:String}, JSONEachRow, 'key String, value UInt64')

Query id: aa0d6fbf-8738-47f5-a365-b89213a023bc

┌─key───┬─value─┐
│ hello │     1 │
└───────┴───────┘

1 row in set. Elapsed: 0.551 sec. 

:) 
```

**Comment by lmangani** [comment]
Thanks @ramazanpolat 

The instructions work fine one 22.6 up to 22.10 in my tests. Did SET change in 22.11 is the question.....

**Comment by ramazanpolat** [comment]
Hmm, checking for 22.9:

```sql
cqmclick1 :) select version()

SELECT version()

Query id: 8c2e7f3b-44fa-440c-8911-bdc633fe8ca5

┌─version()─┐
│ 22.9.3.18 │
└───────────┘

1 row in set. Elapsed: 0.003 sec.

cqmclick1 :) SET custom_url = 'https://urleng.com/4c5u94cw592bsq9wypba'

SET custom_url = 'https://urleng.com/4c5u94cw592bsq9wypba'

Query id: d07a520f-83d7-4f2a-824b-b49a798f0592


0 rows in set. Elapsed: 0.002 sec.

Received exception from server (version 22.9.3):
Code: 115. DB::Exception: Received from localhost:9000. DB::Exception: Unknown setting custom_url. (UNKNOWN_SETTING)
```

**Comment by ramazanpolat** [comment]
And also for other versions as well.
https://fiddle.clickhouse.com/5ff0db17-750a-4056-ab3f-25e873d19f41
````
Received exception from server (version 22.6.9):
Code: 115. DB::Exception: Received from localhost:9000. DB::Exception: Unknown setting custom_url. (UNKNOWN_SETTING)
(query: SET custom_url = 'https://urleng.com/4c5u94cw592bsq9wypba'
)
--
Received exception from server (version 22.7.7):
Code: 115. DB::Exception: Received from localhost:9000. DB::Exception: Unknown setting custom_url. (UNKNOWN_SETTING)
(query: SET custom_url = 'https://urleng.com/4c5u94cw592bsq9wypba'
)
--
Received exception from server (version 22.8.7):
Code: 115. DB::Exception: Received from localhost:9000. DB::Exception: Unknown setting custom_url. (UNKNOWN_SETTING)
(query: SET custom_url = 'https://urleng.com/4c5u94cw592bsq9wypba'
)
--
Received exception from server (version 22.10.1):
Code: 115. DB::Exception: Received from localhost:9000. DB::Exception: Unknown setting custom_url. (UNKNOWN_SETTING)
(query: SET custom_url = 'https://urleng.com/4c5u94cw592bsq9wypba'
)
````

**Comment by lmangani** [comment]
That's .. quite interesting! I cannot reproduce it, it even works on the serverless 22.9 and 22.10 for me and I wonder why...  
I guess we'll change the examples to your suggestion! :)

**Comment by ramazanpolat** [comment]
@lmangani How come you can run this? This is really bugging me : ) Because I've always wanted to be able to set my own variables with SET but unable to do it without prefixing variables by `param_`.

>  it even works on the serverless 22.9 and 22.10 for me

By serverless, do you mean `clickhouse-local`?

**Comment by lmangani** [comment]
Could you check if the PR looks fine? 

> By serverless, do you mean clickhouse-local?

Almost, its a [custom build + wrapper](https://blog.qryn.dev/wrapping-a-serverless-clickhouse-pt2) based off the master ClickHouse branch

**Comment by lmangani** [comment]
> @lmangani How come you can run this? This is really bugging me : ) Because I've always wanted to be able to set my own variables with SET but unable to do it without prefixing variables by `param_`.

_You were right! Mystery solved. It's a custom settings prefix we have as a default on our deployments!_

```
<clickhouse>
    <custom_settings_prefixes>custom_</custom_settings_prefixes>
</clickhouse>
```

**Comment by lmangani** [comment]
Issue should be [fixed](https://urleng.com) - could you confirm this works for you? It doesn't with the `custom_` extension... so i'm not sure

> :) SET param_url = 'https://urleng.com/xxxxxxxxxxxxx'
> :) INSERT INTO FUNCTION url({url:String}, JSONEachRow, 'key String, value UInt64') VALUES ('hello', 1)
> :) SELECT * FROM url({url:String}, JSONEachRow)

**Comment by ramazanpolat** [comment]
Fixed by #2
